modules/platform/nuxeo-automation/nuxeo-automation-scripting/src/main/java/org/nuxeo/automation/scripting/internals/AutomationScriptingServiceImpl.java

```diff
@@ -174,17 +170,20 @@ protected ScriptEngine getScriptEngine() {
     }
 
     protected ScriptEngine getScriptEngine(boolean cache, boolean filter) {
-        NashornScriptEngineFactory nashorn = new NashornScriptEngineFactory();
-        String[] args = cache
-                ? new String[] { "-strict", "--optimistic-types=true", "--persistent-code-cache",
-                        "--class-cache-size=50" }
-                : new String[] { "-strict" };
-        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
-        ClassFilter classFilter = filter ? getClassFilter() : null;
-        return nashorn.getScriptEngine(args, classLoader, classFilter);
+        ScriptEngine engine = new ScriptEngineManager().getEngineByName("JavaScript");
+        if (engine == null) {
+            throw new UnsupportedOperationException(NASHORN_WARN_ENGINE);
+        }
+        Bindings bindings = engine.getBindings(ScriptContext.ENGINE_SCOPE);
+        bindings.put("polyglot.js.allowHostAccess", true);
+        Predicate<String> classFilter = filter ? getClassFilter() : null;
+        if (classFilter != null) {
+            bindings.put("polyglot.js.allowHostClassLookup", classFilter);
```

Comment:
i may need to equip this engine with Nashorn's compatibility mode to make this mergable upstream.